rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2309,6 +2309,77 @@ void getProposedFederationCreationBlockNumber_whenStorageProviderReturnsProposed
         assertEquals(creationBlockNumber, actualCreationBlockNumber.get());
     }
 
+    @Test
+    void getProposedFederatorPublicKeyOfType_whenFederationIsEmpty_shouldReturnEmpty() {
```

Comment:
```suggestion
    @ParameterizedTest
    @ValueSource(KeyType.BTC, KeyType.RSK, KeyType.MST)
    void getProposedFederatorPublicKeyOfType_whenFederationIsEmpty_shouldReturnEmpty() {
```
What do you think? To cover some extra scenarios. Same can be applied to the remaining tests

Reply:
Sounds good, I have updated it. Let me know if it looks fine.